src/Data/List/Relation/Binary/Sublist/Setoid.agda

```diff
@@ -79,3 +79,80 @@ open HeterogeneousProperties.Antisymmetry {R = _≈_} {S = _≈_} (λ x≈y _ 
 ⊆-poset = record
   { isPartialOrder = ⊆-isPartialOrder
   }
+
+------------------------------------------------------------------------
+-- Weak pushout
+--
+-- The category _⊆_ does not have proper pushouts.  For instance consider:
+--
+--   τᵤ : [] ⊆ (u ∷ [])
+--   τᵥ : [] ⊆ (v ∷ [])
+--
+-- Then, there are two unrelated upper bounds (u ∷ v ∷ []) and (v ∷ u ∷ []),
+-- since _⊆_ does not include permutations.
+--
+-- Even though there are no unique least upper bounds, we can merge two
+-- extensions of a list, producing a minimial superlist of both.
+--
+-- For the example, the left-biased merge would produce the pair:
+--
+--   τᵤ′ : (u ∷ []) ⊆ (u ∷ v ∷ [])
+--   τᵥ′ : (v ∷ []) ⊆ (u ∷ v ∷ [])
+--
+-- We call such a pair a raw pushout.  It is then a weak pushout if the
+-- resulting square commutes, i.e.:
+--
+--   ⊆-trans τᵤ τᵤ′ ~ ⊆-trans τᵥ τᵥ′
+--
+-- This requires a notion of equality _~_ on sublist morphisms.
+--
+-- Further, commutation requires a similar commutation property
+-- for the underlying equality _≈_, namely
+--
+--   trans x≈y (sym x≈y) == trans x≈z (sym x≈z)
+--
+-- for some notion of equality _==_ for equality proofs _≈_.
+-- Such a property is given e.g. if _≈_ is proof irrelevant
+-- or forms a groupoid.
+
+private
+  variable
+    x y z : A
+    xs ys zs us vs : List A
+    τ τ′ τ₁ τ₂ σ σ′ : xs ⊆ ys
```

Comment:
Standard library convention is that we're not currently using variables for anything except levels and sets.

Reply:
Just saw this in the style guide.    But it seems to contradict https://github.com/agda/agda-stdlib/pull/841#pullrequestreview-262429559 where the opposite advice was given.

Reply:
If you want to stick to the strict `variable` policy, I can of course give the explicit type signatures, no problem.

Reply:
I'm pretty sure @gallais gave the same advice in #841 

> I think this is a good opportunity to add a private variable block at
the top of Data.List.Relation.Binary.Sublist.Heterogeneous.Properties
declaring the **levels & sets** used in all of these modules.

As for a rationale, It's pretty much for the same reason as you mentioned earlier. The more complex interactions between variables is still a little unstable so we don't want to dive into using them too much for the moment.

Reply:
I removed this variable block now.